Initial upload of NGE docs

PatrickRyanMS opened the pull request:
Initial draft of NGE pages MD as well as support images.

closes #952

deltakosh commented:
Ok I will do it :)

PatrickRyanMS committed:
Initial upload of NGE docs

Initial draft of NGE pages MD as well as support images.

deltakosh reviewed (commented):
some questions ;)

RaananW reviewed (approved):
My review is technical (documentation) - LGTM.

Makes me wonder if I need to extend the nme support with nge as well? we have NME support in the playground - you can embed NME snippets. I can add support for NGE as well.

PatrickRyanMS committed:
Update ngeBlocks.md

update description of optimize node with information about properties that are discarded in the operation

PatrickRyanMS committed:
Updates from PR comments

Updates to add language about shortcuts and a change in the behavior of debug.

PatrickRyanMS committed:
Update instances nodes descriptions

Add information to the docs that specifically states that instance node create the specified number of non-null instances so that users aren't confused if they introduce random nulls into instance nodes.

Popov72 reviewed (approved):
Great explanations! It could practically be reused "as is" for the NME!

Only a couple of nits from me.

[edit] Oups, I noticed I forgot to review the NGE blocks page! I will create another review if needed.

PatrickRyanMS committed:
Add context for random while looping

The loop node needed some extra context about how it works when used within loops and nested loops. Added Lock per loop advanced property info

PatrickRyanMS committed:
Changes from Alexis's Review

Grammar, spelling, and clarification edits from Alexis's review

PatrickRyanMS commented:
@Popov72, I was writing the general tool doc with the intention to reuse much of it to update the NME general tool doc. The NME nodes also need the same expanded treatment that was done for NGE. It will take some time as it did with the NGE nodes because a lot of testing is done for each node to ensure we cover everything. Good news is that bugs get uncovered that way too.

Popov72 reviewed (commented):
WoW, this NGE blocks page is quite a beast!

I have mostly nits for this one.

(I think you are aware of it, but just as a reminder, a few images have to be changed and the `Align` block is missing the documentation)

PatrickRyanMS committed:
Updates from Alexis's second review + matrix compose

These are updates from Alexis's second review as well as a full spell check. Also completed the Align entry as well as updated the AlignNode binary to push up again as it did not appear to commit with the original PR. Also added the Matrix Compose docs as it's a new node that has come in since the original PR and it seemed logical to add it now.

PatrickRyanMS committed:
Merge remote-tracking branch 'upstream/master'

PatrickRyanMS committed:
Update node anchor links with new tag

Update to the tableOfContentItem component with direction from Raanan to fix a missing line in the H3 tag code.

Update the nodes doc to exchange the image and h3 tag with the new H3 tag to allow anchor linking to the image rather than to the title of the node which scrolls the image of the node off the top of the screen.

PatrickRyanMS commented:
@Popov72, the changes to the anchor links from @RaananW's addition of the new H2 and H3 image anchor links is in for commit https://github.com/BabylonJS/Documentation/pull/963/commits/81ddf999ffd065a130437ae1d3214eceb956179e 

I can't find the comment where we were talking about this earlier, but they should be all in and working.

PatrickRyanMS committed:
Corrections to ArcXXX node desctiptions.

Missed this comment before. Removed error in unit description in the ArcXXX nodes descriptions.
